src/Orc.NuGetExplorer/Orc.NuGetExplorer.Shared/Models/PackageDetails.cs

```diff
@@ -73,15 +94,332 @@ public string Dependencies
         public bool? IsInstalled { get; set; }
 
         public string FullName { get; }
+
         public string Description { get; }
+
         public Uri IconUrl { get; }
+
         internal IPackage Package { get; }
+
         public DateTime? Published { get; }
+
         public Version Version { get; }
+
         public string SpecialVersion { get; }
+
         public bool IsAbsoluteLatestVersion { get; }
+
         public bool IsLatestVersion { get; private set; }
+
         public bool IsPrerelease { get; }
         #endregion
+
+        #region Methods
+        public void BeginValidation()
```

Comment:
This is all way too complex. Just expose a single IValidationContext property on the package.